selfdrive/selfdrived/selfdrived.py

```diff
@@ -312,13 +312,12 @@ def update_events(self, CS):
           self.events.add(EventName.cameraFrameRate)
     if not REPLAY and self.rk.lagging:
       self.events.add(EventName.selfdrivedLagging)
-    if not self.sm.valid['radarState']:
-      if self.sm['radarState'].radarErrors.canError:
-        self.events.add(EventName.canError)
-      elif self.sm['radarState'].radarErrors.radarUnavailableTemporary:
-        self.events.add(EventName.radarTempUnavailable)
-      else:
-        self.events.add(EventName.radarFault)
+    if self.sm['radarState'].radarErrors.canError:
+      self.events.add(EventName.canError)
+    elif self.sm['radarState'].radarErrors.radarUnavailableTemporary:
+      self.events.add(EventName.radarTempUnavailable)
+    elif any(getattr(self.sm['radarState'].radarErrors, f) for f in self.sm['radarState'].radarErrors.schema.fields):
+      self.events.add(EventName.radarFault)
```

Comment:
You can also check if liveTracks is valid. There's a simpler way to write this last check:

https://github.com/commaai/openpilot/blob/bdd6ff4f3e6fac235995d825d747081d75d13407/selfdrive/car/card.py#L224

Reply:
Yeah, I think if you swap the original for this, it should work:

```
if not self.sm.valid['liveTracks']:
```

Or keep it checking radarState, but check modelV2 still valid